backend/overstroomik_service/pdok.py

```diff
@@ -0,0 +1,143 @@
+"""
+This class uses the pdok-api to find address information
+
+"""
+
+from typing import Optional
+from pydantic import BaseModel
+from overstroomik_service.auto_models import Data, FloodInfo, Webservice, Location
+from overstroomik_service.config import settings
+import httpx
+
+
+class PDOK():
+
+    # api url (exampl:https://geodata.nationaalgeoregister.nl/locatieserver/v3/free)
+    api = settings.PDOK_API
+    fields = "centroide_rd,centroide_ll,straatnaam,woonplaatsnaam,postcode"
+
+    async def get_address(self, 
+                          searchfield: Optional[str], 
+                          latitude: Optional[float], 
+                          longitude: Optional[float]):
+        """
+        Search address information from the PDOK services.
+        """
+
+        adress_item = {}
+
+        # start with no errors
+        status = settings.ERROR_GENERAL_NOER
+
+        # search by text or by lat/lon
+        if searchfield is not None and len(searchfield.strip()) > 0:
+            status, adress_item = await self.address_by_searchfield(searchfield)
+        elif latitude and longitude:
+            status, adress_item = await self.address_by_latlon(latitude, longitude)
+        else:
+            status = settings.ERROR_GENERAL_ERRO
+
+        # return status and location
+        return status, Location(
+            search_field=searchfield,
+            latitude=adress_item.get("latitude", latitude),
```

Comment:
Why not create Location in the address_ function? You could even use **kwargs here.